test/expect-tests/fiber/fiber_tests.ml

```diff
@@ -792,7 +803,6 @@ let%expect_test "Stream: multiple writers is an error" =
   Trailing output
   ---------------
   Writer 1 writing
-  Reader 1 done
```

Comment:
The old behaviour was wrong: we were executing a job from a previous `Fiber.run`. I added a assertion in `fiber.ml` to make sure this cannot happen anymore.

Makes me thinks that `Ivar.create` and co should return a `Fiber.t`, so that we can't create an ivar outside of `Fiber.run`, which is what led to this issue.